searcher/analyse/analyse.go

```diff
@@ -0,0 +1,61 @@
+package analyse
+
+import (
+	"log"
+	"strings"
+
+	"github.com/sleepypioneer/resource-searcher/searcher/resource"
+)
+
+var (
+	keywords    = []string{"python", "webservice", "server", "Django", "list"}
+	punctuation = []string{".", ",", "!", ":", ";", "'", "/", "-", "_", "*", "&", "{", "}", "[", "]"}
+)
+
+func stringInSlice(a string, list []string) bool {
+	for _, b := range list {
+		if b == a {
+			return true
+		}
+	}
+	return false
+}
+
+// WordCount returns total number of words in the article
+func WordCount(content []string) int {
+	wordCount := 0
+	// for each paragraph in the article we count how many words it contains.
+	for _, p := range content {
+		// remove all punction in paragraph
+		for _, punc := range punctuation {
+			p = strings.ReplaceAll(p, punc, "")
+		}
+		// split paragraph into words list
+		words := strings.Split(p, " ")
+		wordCount += len(words)
+	}
+	// return total word count for the article
+	return wordCount
+}
+
+// FindTopics retuns a list of topics which the article includes
+func FindTopics(content []string) (topics []string) {
```

Comment:
You can greatly improve the speed of this function. In general, a loop inside a loop is a sign that something *could* be optimised. It's not always the case, by far! But it's always a good place to stop and think a bit about efficiency: Am I computing something multiple times? Are most iterations without impact? Could a different data model speed up the iteration? 

Let's use this case to create a [benchmark](https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go) for 2 or more implementation of the same `FindTopics` function, which is basically just the intersection of both sets of words, right?

Ideas:
* Use a `map[string]struct{}` instead of a slice for the keywords
* Use a `map[string]struct{}`instead of a slice for the resulting topics
* Iterate over the shorter of both sets of words to find the matches
* ...?

Reply:
So Benchmark with current function: 
`ok      github.com/sleepypioneer/resource-searcher/searcher/analyse     2.398s`
  
Function that breaks up loops and temporily stores all words in content in a `[]string` (Commit: 4fa3879ce42bd0e2439e59d1d8bf38ea3cbc8b76 )
`ok      github.com/sleepypioneer/resource-searcher/searcher/analyse     1.106s`

Reply:
> As we perform multiple steps of cleaning for the content, would it be interesting to store that cleaned content within the struct maybe?

I've also looked at this as I thought that may also help speed up my analyse package. See PR: #5 

However the results on the benchmark is higher:
`ok      github.com/sleepypioneer/resource-searcher/searcher/analyse     1.389s`